pandas/io/formats/style.py

```diff
@@ -499,6 +501,49 @@ def format(self, formatter, subset=None, na_rep: Optional[str] = None) -> "Style
                 self._display_funcs[(i, j)] = formatter
         return self
 
+    def set_data_classes(self, classes: DataFrame) -> "Styler":
```

Comment:
this is in-line modifying state on the Styler, is this what we do in other operations? we generally do not what to do this (but it may be that Styler has this pattern).

cc @TomAugspurger

Reply:
Yeah IIRC Styler does a lot of inplace mutation :/

Reply:
No comment on the existing implementation, but I feel this PR and the other Styler PRs I have outstanding are trying to maintain the same architecture.

Reply:
Right, agreed there's nothing to do here. We can clean this all up later if wanted.